README.md
# eta-coser Setup Guide

### Building

Go to the directory of source code and run the fallowing in order to get application binary file:
```bash
cd etacoser
qmake
sudo make
```
Now you should be able to see **etacoser** executable binary file.

### Install

To install the application to the system run the fallowing code in source code directory:

```bash
chmod +x etacoser
sudo cp -Rvf etacoser /usr/bin/.
```
and then go to etacoserscripts directory and run the fallowing code:

```bash
cd ../etacoserscripts
chmod +x setup.sh
./setup.sh
```

### Configuration

If you want to change some settings in application go to **/usr/share/eta-coser** directory and modify config.ini file.


debian/changelog
eta-coser (0.2.3) stable; urgency=medium

  * Update control file
  * Pardus CI integration

 -- Yunusemre Şentürk <yunusemre.senturk@pardus.org.tr>  Thu, 13 Sep 2018 14:04:36 +0300

eta-coser (0.2.2) unstable; urgency=medium

  * Clear up white spaces and remove any kdm related system cb

 -- Yunusemre Şentürk <yunusemre.senturk@pardus.org.tr>  Tue, 17 Oct 2017 10:18:53 +0300

eta-coser (0.2.1) unstable; urgency=medium

  * Redesign for lightdm and get rid of kdm

 -- Yunusemre Şentürk <yunusemre.senturk@pardus.org.tr>  Tue, 17 Oct 2017 09:52:48 +0300

eta-coser (0.2.0) unstable; urgency=medium

  * Initial release.

 -- Yunusemre Şentürk <yunusemre.senturk@pardus.org.tr>  Thu, 14 Sep 2017 14:31:37 +0300



main.cpp
/*****************************************************************************
 *   Copyright (C) 2016 by Yunusemre Senturk                                 *
 *   <yunusemre.senturk@pardus.org.tr>                                       *
 *                                                                           *
 *   This program is free software; you can redistribute it and/or modify    *
 *   it under the terms of the GNU General Public License as published by    *
 *   the Free Software Foundation; either version 2 of the License, or       *
 *   (at your option) any later version.                                     *
 *                                                                           *
 *   This program is distributed in the hope that it will be useful,         *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of          *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
 *   GNU General Public License for more details.                            *
 *                                                                           *
 *   You should have received a copy of the GNU General Public License       *
 *   along with this program; if not, write to the                           *
 *   Free Software Foundation, Inc.,                                         *
 *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .          *
 *****************************************************************************/

#include "eta-coser.h"
#include <QApplication>

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);
    EtaCoser w;
    w.show();

    return a.exec();
} 


eta-coser.h
/*****************************************************************************
 *   Copyright (C) 2016 by Yunusemre Senturk                                 *
 *   <yunusemre.senturk@pardus.org.tr>                                       *
 *                                                                           *
 *   This program is free software; you can redistribute it and/or modify    *
 *   it under the terms of the GNU General Public License as published by    *
 *   the Free Software Foundation; either version 2 of the License, or       *
 *   (at your option) any later version.                                     *
 *                                                                           *
 *   This program is distributed in the hope that it will be useful,         *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of          *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
 *   GNU General Public License for more details.                            *
 *                                                                           *
 *   You should have received a copy of the GNU General Public License       *
 *   along with this program; if not, write to the                           *
 *   Free Software Foundation, Inc.,                                         *
 *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .          *
 *****************************************************************************/

#ifndef etacoser_H
#define etacoser_H

#include <QMainWindow>

class QLabel;
class QTimer;

class EtaCoser : public QMainWindow
{
    Q_OBJECT

public:
    EtaCoser();
    virtual ~EtaCoser();
private:
    bool startwithpardusflag;
    int time;
    QLabel * statuslabel;
    QTimer *countdown;
    QString operatingsystem;
private slots:
    void startWithWindows();
    void startWithPardus();
    void timerShotCallBack();
    void setStatusLabel(int seconds);
    void setTime();
};

#endif // etacoser_H


etacoser/etacoser.cpp
#include "etacoser.h"

#include <QtGui/QPushButton>
#include <QtGui/QDesktopWidget>
#include <QtGui/QHBoxLayout>
#include <QtGui/QIcon>
#include <QtGui/QPalette>
#include <QSettings>

EtaCoser::EtaCoser()
{
    QSettings settings("/usr/share/eta-coser/config.ini",QSettings::IniFormat);
    settings.beginGroup("eta-coser");
    operatingsystem = settings.value("Start").toString();
    if(operatingsystem=="Pardus")
    {
        startwithpardusflag = true;
    }
    else
    {
        startwithpardusflag = false;
    }

    time = settings.value("Time").toInt();
    settings.endGroup();

    QDesktopWidget dw;
    int sw = dw.screenGeometry(dw.primaryScreen()).width();
    int sh = dw.screenGeometry(dw.primaryScreen()).height();
    
    setWindowFlags(Qt::X11BypassWindowManagerHint);
    resize(sw,sh);
    
    QPushButton* btnPardus = new QPushButton(this);
    QPushButton* btnWindows = new QPushButton(this);
    
    btnPardus->setSizePolicy(QSizePolicy::Expanding,QSizePolicy::Expanding);
    btnWindows->setSizePolicy(QSizePolicy::Expanding,QSizePolicy::Expanding);
    
    btnWindows->setIcon(QIcon("/usr/share/eta-coser/windows-logo.png"));
    btnWindows->setIconSize(QSize(328,337));
    btnPardus->setIcon(QIcon("/usr/share/eta-coser/pardus-logo.png"));
    btnPardus->setIconSize(QSize(304,373));

    btnPardus->setStyleSheet( "background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1,stop:0 #ffcc00, stop:1 #ff9c00);");
    
    btnWindows->setStyleSheet( "background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1,stop:0 #56c4ff, stop:1 #006ac0);");

    QWidget* centralWidget = new QWidget(this);
    centralWidget->setSizePolicy(QSizePolicy::Expanding,QSizePolicy::Expanding);
    centralWidget->setGeometry(0,0,sw,sh-50);
    
    QHBoxLayout* layout = new QHBoxLayout(centralWidget);
    layout->addWidget(btnPardus);
    layout->addWidget(btnWindows);

    statuslabel = new QLabel(this);
    statuslabel->setGeometry(0,sh-50,sw,50);
    statuslabel->setStyleSheet("QLabel{color: black; qproperty-alignment: AlignCenter;}");
    QFont f;
    f.setPointSize(30);
    statuslabel->setFont(f);

    countdown = new QTimer(this);
    connect(countdown,SIGNAL(timeout()),this,SLOT(setTime()));
    connect(btnPardus,SIGNAL(clicked()),this,SLOT(startWithPardus()));
    connect(btnWindows,SIGNAL(clicked()),this,SLOT(startWithWindows()));

    QTimer::singleShot(time*1000,this,SLOT(timerShotCallBack()));

    countdown->start(1000);
    setStatusLabel(time--);
}

EtaCoser::~EtaCoser()
{}

void EtaCoser::startWithWindows()
{
    system("sudo grub-reboot 2 && reboot now");
}

void EtaCoser::startWithPardus()
{
    this->close();
}

void EtaCoser::timerShotCallBack()
{
    if(startwithpardusflag)
    {
        startWithPardus();
    }
    else
    {
        startWithWindows();
    }

}

void EtaCoser::setTime()
{
    if(time == 0)
    {
        countdown->stop();
    }
    else
    {
        setStatusLabel(time--);
    }
}

void EtaCoser::setStatusLabel(int seconds)
{
    QString output = operatingsystem+ QString::fromUtf8(" Açılıyor ")+ QString::number(seconds);
    statuslabel->setText( output);
}


eta-coser.cpp
/*****************************************************************************
 *   Copyright (C) 2016 by Yunusemre Senturk                                 *
 *   <yunusemre.senturk@pardus.org.tr>                                       *
 *                                                                           *
 *   This program is free software; you can redistribute it and/or modify    *
 *   it under the terms of the GNU General Public License as published by    *
 *   the Free Software Foundation; either version 2 of the License, or       *
 *   (at your option) any later version.                                     *
 *                                                                           *
 *   This program is distributed in the hope that it will be useful,         *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of          *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
 *   GNU General Public License for more details.                            *
 *                                                                           *
 *   You should have received a copy of the GNU General Public License       *
 *   along with this program; if not, write to the                           *
 *   Free Software Foundation, Inc.,                                         *
 *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .          *
 *****************************************************************************/

#include "eta-coser.h"

#include <QPushButton>
#include <QDesktopWidget>
#include <QHBoxLayout>
#include <QIcon>
#include <QPalette>
#include <QSettings>
#include <QLabel>
#include <QTimer>

EtaCoser::EtaCoser()
{
    QSettings settings("/usr/share/eta/eta-coser/config.ini",
                       QSettings::IniFormat);
    settings.beginGroup("eta-coser");
    operatingsystem = settings.value("Start").toString();
    if(operatingsystem=="Pardus") {
        startwithpardusflag = true;
    } else {
        startwithpardusflag = false;
    }

    time = settings.value("Time").toInt();
    settings.endGroup();

    QDesktopWidget dw;
    int sw = dw.screenGeometry(dw.primaryScreen()).width();
    int sh = dw.screenGeometry(dw.primaryScreen()).height();

    setWindowFlags(Qt::X11BypassWindowManagerHint);
    resize(sw,sh);

    QPushButton* btnPardus = new QPushButton(this);
    QPushButton* btnWindows = new QPushButton(this);

    btnPardus->setSizePolicy(QSizePolicy::Expanding,QSizePolicy::Expanding);
    btnWindows->setSizePolicy(QSizePolicy::Expanding,QSizePolicy::Expanding);

    btnWindows->setIcon(QIcon(":images/windows-logo.png"));
    btnWindows->setIconSize(QSize(328,337));
    btnPardus->setIcon(QIcon(":images/pardus-logo.png"));
    btnPardus->setIconSize(QSize(304,373));

    btnPardus->setStyleSheet( "background-color: qlineargradient(x1:0, y1:0, \
                              x2:0, y2:1,stop:0 #ffcc00, stop:1 #ff9c00);");

    btnWindows->setStyleSheet( "background-color: qlineargradient(x1:0, y1:0, \
                               x2:0, y2:1,stop:0 #56c4ff, stop:1 #006ac0);");

    QWidget* centralWidget = new QWidget(this);
    centralWidget->setSizePolicy(QSizePolicy::Expanding,QSizePolicy::Expanding);
    centralWidget->setGeometry(0,0,sw,sh-50);

    QHBoxLayout* layout = new QHBoxLayout(centralWidget);
    layout->addWidget(btnPardus);
    layout->addWidget(btnWindows);

    statuslabel = new QLabel(this);
    statuslabel->setGeometry(0,sh-50,sw,50);
    statuslabel->setStyleSheet("QLabel{color: black; qproperty-alignment: \
                               AlignCenter;}");
    QFont f;
    f.setPointSize(30);
    statuslabel->setFont(f);

    countdown = new QTimer(this);
    connect(countdown,SIGNAL(timeout()),this,SLOT(setTime()));
    connect(btnPardus,SIGNAL(clicked()),this,SLOT(startWithPardus()));
    connect(btnWindows,SIGNAL(clicked()),this,SLOT(startWithWindows()));

    QTimer::singleShot(time*1000,this,SLOT(timerShotCallBack()));

    countdown->start(1000);
    setStatusLabel(time--);
}

EtaCoser::~EtaCoser()
{}

void EtaCoser::startWithWindows()
{
    system("sudo grub-reboot 2 && reboot now");
}

void EtaCoser::startWithPardus()
{
    this->close();
}

void EtaCoser::timerShotCallBack()
{
    if(startwithpardusflag) {
        startWithPardus();
    } else {
        startWithWindows();
    }
}

void EtaCoser::setTime()
{
    if(time == 0) {
        countdown->stop();
    } else {
        setStatusLabel(time--);
    }
}

void EtaCoser::setStatusLabel(int seconds)
{
    QString output = operatingsystem + QString::fromUtf8(" Açılıyor ") +
            QString::number(seconds);
    statuslabel->setText( output);
}
